metadata-db-app/src/cmr/metadata_db/data/memory_db.clj

```diff
@@ -10,35 +10,64 @@
    [cmr.metadata-db.data.concepts :as concepts]
    [cmr.metadata-db.data.ingest-events :as ingest-events]
    [cmr.metadata-db.data.oracle.concepts.tag :as tag]
+   [cmr.metadata-db.data.oracle.concepts.variable :as variable]
    [cmr.metadata-db.data.oracle.concepts]
    [cmr.metadata-db.data.providers :as providers]
    [cmr.metadata-db.services.provider-validation :as pv]))
 
-(defn after-save
-  "Handler for save calls. It will be passed the list of concepts and the concept that was just
-  saved. It should manipulate anything required and return the new list of concepts."
+(defmulti after-save
```

Comment:
This is what I had in mind on how to refactor the `after-save` function:
{code}
(defn after-save
  "Handler for save calls. It will be passed the list of concepts and the concept that was just
  saved. It should manipulate anything required and return the new list of concepts."
  [db concepts concept]
  (cond (and (= :collection (:concept-type concept))
             (:deleted concept))
    (filter #(not= (:concept-id concept) (get-in % [:extra-fields :parent-collection-id]))
            concepts)
    
    ;; CMR-2520 Remove this case when asynchronous cascaded deletes are implemented.
    (and (or (= :tag (:concept-type concept))
             (= :variable (:concept-type concept)))
         (:deleted concept))
    (let [associations (tag/get-associations-for-tombstone db concept)
          tombstones (map (fn [ta] (-> ta
                                       (assoc :metadata "" :deleted true)
                                       (update :revision-id inc)))
                          associations)]
      ;; publish tag-association delete events
      (doseq [tombstone tombstones]
        (ingest-events/publish-event
         (:context db)
         (ingest-events/concept-delete-event tombstone)))
      (concat concepts tombstones))
    
    :else concepts))
{code}
Where `tag/get-associations-for-tombstone` is the multimethod that replaces `get-tag-associations-for-tag-tombstone` and the one for Variable.